common/test/acceptance/tests/lms/test_lms_problems.py

```diff
@@ -319,3 +327,81 @@ def test_partial_credit(self):
         problem_page.fill_answer_numerical('-1')
         problem_page.click_check()
         self.assertTrue(problem_page.simpleprob_is_partially_correct())
+
+
+class LogoutDuringAnswering(ProblemsTest):
+    """
+    Tests for the scenario where a user is logged out (their session expires
+    or is revoked) just before they click "check" on a problem.
+    """
+    def get_problem(self):
+        """
+        Create a problem.
+        """
+        xml = dedent("""
+            <problem>
+                <p>The answer is 1</p>
+                <numericalresponse answer="1">
+                    <formulaequationinput label="where are the songs of spring?" />
+                    <responseparam type="tolerance" default="0.01" />
+                </numericalresponse>
+            </problem>
+        """)
+        return XBlockFixtureDesc('problem', 'TEST PROBLEM', data=xml)
+
+    def log_user_out(self):
+        """
+        Log the user out by deleting their session cookie.
+        """
+        self.browser.delete_cookie('sessionid')
+
+    def test_logout_after_click_redirect(self):
+        """
+        1) User goes to a problem page.
+        2) User fills out an answer to the problem.
+        3) User is logged out because their session id is invalidated or removed.
+        4) User clicks "check", and sees a confirmation modal asking them to
+           re-authenticate, since they've just been logged out.
+        5) User clicks "ok".
+        6) User is redirected to the login page.
+        7) User logs in.
+        8) User is redirected back to the problem page they started out on.
+        """
+        self.courseware_page.visit()
+        problem_page = ProblemPage(self.browser)
+        self.assertEqual(problem_page.problem_name, 'TEST PROBLEM')
+        problem_page.fill_answer_numerical('1')
+        self.log_user_out()
+        with problem_page.handle_alert(confirm=True):
+            problem_page.click_check()
+            problem_page.wait_for_ajax()
+
+        login_page = LoginPage(self.browser)
+        login_page.wait_for_page()
+
+        login_page.login(self.EMAIL, self.PASSWORD, expected_redirect=ProblemPage)
```

Comment:
Yeah, I wasn't sure what the right move was here — I wanted it to be clear that we were on the same page that we were on in the beginning of the test. But perhaps 
```python
self.assertEqual(problem_page.problem_name, 'TEST PROBLEM')
```
accomplishes that? What's your thought?